stroeer/core/v1/article.proto

```diff
@@ -8,133 +8,317 @@ option go_package = "github.com/stroeer/tapir/go/stroeer/core/v1;core";
+  // was updated in seconds of UTC time since Unix epoch.
   google.protobuf.Timestamp updated = 7;
 }
 
+// Elements of the article which are not part of the body, e.g. author information
+// opener image/video etc. Those elements should be used to render the article
+// as a teaser e.g. on overview pages.
+//
+// Different types of elements like images or videos share
+// the same message structure distinguished by the ElementType field.
 message Element {
+  // Type of the element.
   ElementType type = 1;
+
+  // TODO: this might be obsolete with https://github.com/orgs/stroeer/teams/api/discussions/1
   repeated ElementRelation relations = 2;
+
+  // Assets of the element like images, videos or metadata (required).
   repeated Asset assets = 3;
+
+  // Children of the element.
   repeated Element children = 4;
 }
 
+// Asset of an element.
 message Asset {
+  // Type of the asset.
   AssetType type = 1;
+
+  // TODO: do we still need the index?
   int32 index = 2;
+
+  // Generic map containing general content and configuration information of
+  // the asset.
+  //
+  // Depending on the type of the asset, this map will contain different
+  // entry sets, e.g. 'source', 'width' and 'height' for images.
+  //
+  // Clients must be resilient to unknown or missing entry sets.
   map<string, string> fields = 3;
+
+  // Technical metadata of the asset.
   Metadata metadata = 4;
 }
 
+// Hierarchical section information of the article till root.
+//
+// For example:  /politik/deutschland/ -> /politik/ -> /
 message Section {
+  // Section path of the article, e.g. /politik/deutschland/ (required).
   string path = 1;
+
+  // Label of the section path, e.g. "Politik".
   string label = 2;
+
+  // Parent section. Usually points to the parent section, if there is none, e.g. when in `root` this value is `null`
   Section parent = 3;
 }
 
+// References to get access to the article on different technical platforms like
+// canonical web or AMP.
 message References {
-  // article's path, e.g. /section/id_$ID/title.html
+  // URL path for this article e.g. /section/id_$ID/title.html (required).
   string path = 1;
-  // article's canonical URL, may differ if external, e.g. https://www.giga.de/external.html
+
+  // Canonical URL of the article, may differ if external, e.g. https://www.giga.de/external.html
+  // (required).
   string canonical = 2;
-  // article's AMP URL, no pattern specified yet.
+
+  // AMP URL of the article.
   string amp = 3;
 }
 
+// Body of the article to render on detail pages.
 message Body {
+  // Recursive Body nodes.
   repeated BodyNode children = 1;
 }
 
+// Recursive body structure representing all types of possible
+// body nodes inside an article.
+//
+// Clients must be resilient to unknown or missing nodes.
 message BodyNode {
+  // Type of the node, e.g. 'a', 'p', 'ul', 'li', 'text' (required).
   string type = 1;
+
+  // Text of the node, only set for text nodes (`type == 'text'`).
```

Comment:
Just for clarification: the grpc null  value (== empty string) is still a valid body node? Do not filter body (type text) nodes with empty strings? Does this makes sense? cc @thisismana

Reply:
As for gRPC, `""` is the default value, there is no `null`.
When transforming stuff from the CMS, I have options to strip out the whole `Node` if there is a leaf-node with blank text.